near/kycdao-ntnft/src/lib.rs

```diff
@@ -87,18 +151,22 @@ impl KycdaoNTNFT {
 
         // Get prefilled metadata, also remove digest so it cannot be used again
         let metadata = self.authorized_token_metadata.remove(&digest).expect("Unauthorized code");
+        let status = self.authorized_statuses.remove(&digest).unwrap_or_default();
 
         let token_id = self.next_token_id;
         self.next_token_id = self.next_token_id.checked_add(1).expect("Token ID overflow");
         let token_id_str = token_id.to_string();
 
-        self.tokens.internal_mint(token_id_str, dst, Some(metadata), None)
+        let token = self.tokens.internal_mint(token_id_str.clone(), dst, Some(metadata), None);
+        self.token_statuses.insert(&token_id_str, &status);
+
+        token
     }
 
     // TODO this has a storage cost!
     // TODO add verification path?
     /// @dev Authorize the minting of a new token
-    pub fn authorize_minting(&mut self, auth_code: MintAuthorizationCode, dst: AccountId, metadata: TokenMetadata) {
+    pub fn authorize_minting(&mut self, auth_code: MintAuthorizationCode, dst: AccountId, metadata: TokenMetadata, status: Option<Status>) {
```

Comment:
Slight difference here with the EVM implementation - the entire Status is passed in here; in EVM I only pass the expiry, as I assume we're not minting a revoked token.

Reply:
I thought it's not an issue because the status has a Default implementation, so `is_revoked` can be omitted and will default to false. But now that I think about it, it would be better to use the same interface as Solidity does, it's a good idea to explicitly set the is_revoked to false every time.